discovery-provider/src/monitors/monitoring_queue.py

```diff
@@ -41,6 +42,7 @@ def refresh(redis, db, monitor):
 
 
 @celery.task(name="monitoring_queue", bind=True)
+@save_duration_metric(metric_group="celery_task")
```

Comment:
how confident are you about this not introducing regressions?

Reply:
Fairly confident since we're merely holding onto the `result`, saving a metric, then returning the `result`.

Here is the bulk of the actionable code:

```
            try:
                result = func(*args, **kwargs)
                metric.save_time({"func_name": func.__name__, "success": True})
                return result
            except:
                metric.save_time({"func_name": func.__name__, "success": False})
                raise
```